src/main/scala/outwatch/Sink.scala

```diff
@@ -73,19 +74,42 @@ object Sink {
     * @param next the function to be executed on every emission
     * @param error the function to be executed on error
     * @param complete the function to be executed on completion
-    * @tparam T the type parameter of the consumed elements.
+    * @tparam A the type parameter of the consumed elements.
     * @return a Sink that consumes elements of type T.
     */
-  def create[T](next: T => Future[Ack],
-                error: Throwable => Unit = _ => (),
-                complete: () => Unit = () => ()
-               )(implicit s: Scheduler): IO[Sink[T]] = {
-    IO{
+  def createFull[F[+_], A](next: A => F[Unit],
+                error: Throwable => F[Unit],
```

Comment:
Why do `next` and `complete` need to return `F[_]`?

Reply:
I'd argue this makes them more reusable, in a pure application your effects are going to be in some kind of `F`, so taking an `F`, is probably much easier than having to convert it somehow.